### PR TITLE
sandbox_stop: Call CNI stop before stopping pod infra container

### PR DESCRIPTION
**Pull request by mrunalp**
The CNI hook expects to be able to access the network namespace of
the pod infra container.

@rajatchopra @runcom PTAL

Signed-off-by: Mrunal Patel <mrunalp@gmail.com>

**- What I did**
Moved stop pod infra container after network stop

**- How to verify it**
e2e pass

**- Description for the changelog**

### CODE REVIEW

**Comment by wking** [comment]
I wonder if this is related to #1424, since it's removing a network-namespace action access during network-namespace teardown.  It still seems like the CI kernels are buggy, but perhaps this is the use-case that's landing in the bug.

**Comment by mrunalp** [comment]
@wking Yeah, that is likely. Results will soon tell and we can run rhel_integration another time if it passes on the first run.

**Comment by wking** [comment]
Or maybe this has exposed [an even worse race][1]?  :p

[1]: https://travis-ci.org/kubernetes-incubator/cri-o/jobs/350082692#L3964

**Comment by mrunalp** [comment]
Pushed update. I had missed the state check around calling pod infra stop.

**Comment by mrunalp** [comment]
We need to pick my network lifecycle mgmt into master with this.

**Comment by rhatdan** [comment]
LGTM
@runcom PTAL

**Comment by rhatdan** [comment]
If this does fix https://github.com/kubernetes-incubator/cri-o/issues/1424, lets get it merged...

**Comment by TomSweeneyRedHat** [comment]
LGTM and happy tests.  Possible comment tweak at some point in time.

File: server/sandbox_stop.go [hunk 1]
```diff
@@ -71,7 +71,15 @@ func (s *Server) StopPodSandbox(ctx context.Context, req *pb.StopPodSandboxReque
 	}
 
 	// Clean up sandbox networking and close its network namespace.
+	// We first call network stop before stopping the pod infra container as
+	// stop code accessing the network namespace of pod infra container.
```
[the remark this answers is not in the document]
Comment (wking) [reply server/sandbox_stop.go]: Maybe something like:

> Cleanup the pod network before deleting the sandbox container (after which the kernel will start tearing that network namespace down).

?

**Comment by mrunalp** [comment]
Updated.

**Comment by runcom** [comment]
LGTM

**Comment by mrunalp** [comment]
@rhatdan This is ready to be merged.

**Comment by rhatdan** [comment]
All Green Merging.

**Pull request merged**